packages/zoltan2/src/algorithms/Zoltan2_Algorithm.hpp

```diff
@@ -115,6 +116,91 @@ class Algorithm {
     Z2_THROW_NOT_IMPLEMENTED
   }
 
+  //! \brief  for partitioning methods, return partition tree of the
+  //          computed parts
+  //
+  virtual const std::vector<partitionTreeNode<part_t> > &
+  getPartitionTreeNodes() const
+  {
+    return partitionTree;
+  }
+
+  //! \brief  for partitioning methods, set partition tree of the
+  //          computed parts.
+  //
+  void setPartitionTreeNodes(
```

Comment:
I am surprised that the implementation of these methods is in the base Algorithm class. Other similar methods (e.g., boxAssign) are implemented in the specific algorithms in ways that are appropriate to the specific algorithms; PartitioningSolution then provides a call-through to the algorithm-specific implementation.  Indeed, some of the comments added to this file indicate one might do these operations differently for RCB than for other partitioners.